src/Storefront/Resources/app/storefront/src/plugin-system/plugin.manager.js

```diff
@@ -266,18 +266,65 @@ class PluginManagerSingleton {
         }
 
         if (typeof selector === 'string') {
-            selector = document.querySelectorAll(selector);
+            selector = PluginManagerSingleton._queryElements(selector);
         }
 
         return Iterator.iterate(selector, el => {
             PluginManagerSingleton._initializePluginOnElement(el, pluginClass, options, pluginName);
         });
     }
 
+    /**
+     * Determs the way to query the elements.
+     *
+     * [data-*] => querySelectorAll
+     * #fooBar => getElementById
+     * #foo_bar => getElementById
+     * #foo-bar => getElementById
+     * #foo .bar => querySelectorAll
+     * .fooBar => getElementsByClassName
+     * .FOO => getElementsByClassName
+     * .foo_bar => getElementsByClassName
+     * .foo-bar => getElementsByClassName
+     * .foo .bar => querySelectorAll
+     *
+     * FOO => getElementsByTagName
+     * FOO .bar => querySelectorAll
+     *
+     * For performance reason used regex based on common characters `a-zA-Z1-9_-`
+     * instead of the entire compatible characters
+     *
+     * @param {string} selector
+     *
+     * @return {NodeList|HTMLCollection|Array}
+     */
+    static _queryElements(selector) {
+        // While the use of attributes is the most common right now, we should return early.
+        if (selector.startsWith('[')) {
+            return document.querySelectorAll(selector);
+        }
+
+        if (selector.startsWith('.')) {
+            const regexEl = /^\.([\w-]+)$/.exec(selector);
+            if (regexEl) {
+                return document.getElementsByClassName(regexEl[1]);
+            }
+        } else if (selector.startsWith('#')) {
```

Comment:
you can skip the `else` keyword because you return in your `if` block:

```
if (selector.startsWith('.')) {
    // ...
}

if (selector.startsWith('#')) {
    // ...
}
// ...
```

Reply:
I don't think so because I want to avoid checking for another startsWith while regexEl doesn't have a result :-)

Reply:
Okay yes that makes sense